src/lib/core/bidi/directionality.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  EventEmitter,
+  Injectable,
+  Optional,
+  SkipSelf
+} from '@angular/core';
+
+export type Direction = 'ltr' | 'rtl';
+
+/**
+ * The directionality (LTR / RTL) context for the application (or a subtree of it).
+ * Exposes the current direction and a stream of direction changes.
+ */
+@Injectable()
+export class Directionality {
+  value: Direction = 'ltr';
+  public change = new EventEmitter<void>();
```

Comment:
How before the tests i can add the dir attribute to the html and the body elements?